pkg/download/download.go

```diff
@@ -38,6 +39,22 @@ func DownloadBinary(unparsedURL string) (string, error) {
 		binaryDirectory = env.DefaultBinaryDir
 	}
 
+	dir, filepath := filepath.Split(binaryDirectory)
+	if filepath != "" {
+		return "", fmt.Errorf("binary directory is not a directory")
```

Comment:
Just a small nit, the error message could be a bit more clearer, about exactly what is wrong with the directory path: 
"supplied download directory is not a directory path (doesn't end with a slash)" or something similar.